pkg/controller/build/reconciler.go

```diff
@@ -471,6 +535,12 @@ func (b *buildReconciler) createNewMachineOSBuildOrReuseExisting(ctx context.Con
 		return fmt.Errorf("could not instantiate new MachineOSBuild: %w", err)
 	}
 
+	// Add container-config annotation
```

Comment:
suggestion: You can use `metav1.SetMetaDataAnnotation()` to simplify this a little.